google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableInstanceAdminClientIT.java

```diff
@@ -0,0 +1,249 @@
+    for (AppProfile appProfile : client.listAppProfiles(instanceId)) {
+      String appProfileId = appProfile.getId();
+      if (appProfileId.startsWith(staleAPPattern) && staleAPPattern.compareTo(appProfileId) > 0) {
+        // Ideally these are stale appProfile so there shouldn't be need for forced delete
+        client.deleteAppProfile(instanceId, appProfileId);
+      }
+    }
+  }
+
+  @Test
+  public void iamUpdateTest() {
+    Policy policy = client.getIamPolicy(instanceId);
+    assertThat(policy).isNotNull();
+
+    Exception actualEx = null;
+    try {
+      assertThat(client.setIamPolicy(instanceId, policy)).isNotNull();
+    } catch (Exception iamException) {
+      actualEx = iamException;
+    }
+    assertThat(actualEx).isNull();
+
+    List<String> permissions =
+        client.testIamPermission(
+            instanceId, "bigtable.tables.readRows", "bigtable.tables.mutateRows");
+    assertThat(permissions).hasSize(2);
+  }
+
+  /* Please set createResource=true to run this. */
+  @Test
+  public void instanceCreationDeletionTest() {
+    assume()
+        .withMessage("This run this test please set createBigtableResource=true in properties")
+        .that(System.getProperty("createBigtableResource"))
+        .isEqualTo("true");
+
+    String newClusterId = TEST_CLUSTER_PREFIX + Instant.now().getEpochSecond();
+    String newInstanceId = TEST_INSTANCE_PREFIX + Instant.now().getEpochSecond();
+
+    try {
+      client.createInstance(
+          CreateInstanceRequest.of(newInstanceId)
+              .addCluster(newClusterId, "us-central1-a", 1, StorageType.SSD)
+              .setDisplayName("Fresh-Instance-Name")
+              .addLabel("state", "readytodelete")
+              .setType(Instance.Type.PRODUCTION));
+
+      assertThat(client.exists(newInstanceId)).isTrue();
+
+      client.updateInstance(
+          UpdateInstanceRequest.of(newInstanceId).setDisplayName("Test-Instance-Name"));
+
+      Instance instance = client.getInstance(newInstanceId);
+      assertThat(instance.getDisplayName()).isEqualTo("Test-Instance-Name");
+
+      assertThat(client.listInstances()).contains(instance);
+
+      client.deleteInstance(newInstanceId);
+      assertThat(client.exists(newInstanceId)).isFalse();
+
+      String staleInstanceId = TEST_INSTANCE_PREFIX + Instant.now().minus(1, ChronoUnit.DAYS);
+      for (Instance ins : client.listInstances()) {
+        String insId = ins.getId();
+        if (insId.startsWith(staleInstanceId) && staleInstanceId.compareTo(insId) > 0) {
+          client.deleteInstance(insId);
+        }
+      }
+    } finally {
+      if (client.exists(newInstanceId)) {
+        client.deleteInstance(newInstanceId);
+      }
+    }
+  }
+
+  /* Please set createResource=true to run this. */
+  @Test
+  @Ignore
+  // TODO: Need to fix cluster creation time. On every cluster creation hung up for a large
+  //  amount of time. (I think it waits till data replication finishes).
```

Comment:
This may not happen regularly but yesterday when I used an instance with some 50MB data,  it ran for ~3 min. Now today I ran it with the empty instance(connectors) it finishes in less than 5 seconds.